connect/mirror/src/main/java/org/apache/kafka/connect/mirror/MirrorCheckpointConnector.java

```diff
@@ -150,10 +156,30 @@ private void loadInitialConsumerGroups()
 
     List<String> findConsumerGroups()
             throws InterruptedException, ExecutionException {
-        return listConsumerGroups().stream()
+        List<String> filterGroups = listConsumerGroups().stream()
                 .map(ConsumerGroupListing::groupId)
-                .filter(this::shouldReplicate)
+                .filter(this::shouldReplicateByGroupFilter)
                 .collect(Collectors.toList());
+
+        List<String> checkpointGroups = new LinkedList<>();
+        List<String> irrelevantGroups = new LinkedList<>();
+
+        for (String group : filterGroups) {
+            Set<String> consumeTopics = listConsumerGroupOffsets(group).keySet().stream()
+                    .map(TopicPartition::topic)
+                    .filter(this::shouldReplicateByTopicFilter)
+                    .collect(Collectors.toSet());
+            // As long as a group consumes a topic configured in "topics", it needs to sync for this group.
+            // Although the topic in "topics.exclude" may be consumed in this group, this will be filtered out in the MirrorCheckpointTask.
+            if (consumeTopics.size() > 0) {
+                checkpointGroups.add(group);
+            } else {
+                irrelevantGroups.add(group);
+            }
+        }
+
+        log.debug("The irrelevant synchronous group list after topic filtering is: {}", irrelevantGroups);
```

Comment:
We could make this a bit clearer for users:
```suggestion
        log.debug(
                "Ignoring the following groups which do not have any offsets for topics that are accepted by the topic filter: {}",
                irrelevantGroups
        );
```

Reply:
nice!